testsuite/tests/lib-option/test.ml

```diff
@@ -0,0 +1,119 @@
+(* TEST
+*)
+
+module Option = Stdlib__option (* why ? *)
```

Comment:
@shindere Could you please have a look at this. It seems that the testing framework is not able to compile the this file if I don't add this (same problem in `Result` PR). If you take for example `lib-uchar` this is not needed so I suspect a bootstrap issue but I don't think a bootstrap should be needed if you add a module to the stdlib and it's not used by the compiler itself.

Also it would be nice to report the actual compilation error when you fail to compile the `test.ml` file rather than a simple exit code.

Reply:
@dbuenzli I could run this test without issue with a fresh build of your branch; maybe try cleaning your tree ?

Reply:
(that is, I could run it without the `module Option = ...` line)

Reply:
@nojb how clean ?  somehow with `make clean` doesn't work here.

Reply:
`git clean -fdx`

Reply:
What is the error given by the compiler ?

Reply:
Tried `make distclean`, didn't work but `git clean -fdx` did, thanks ! (in case you care the error was that `Option` module didn't exist)